src/gui/qgsmapcanvas.cpp

```diff
@@ -990,8 +990,39 @@ void QgsMapCanvas::zoomToSelected( QgsVectorLayer *layer )
   }
 
   rect = mapSettings().layerExtentToOutputExtent( layer, rect );
+
+  // zoom in if point cannot be distinguish from others
+  // also check that rect is empty, as it might not in case of multi points
```

Comment:
Thanks for the comment, was just wondering for multipoint layers with singlepoint features.